### ISSUE TITLE
[Payment card / Subscription] Make `AddBillingCardAndRequestPolicyOwnerChange` 1:1:1

### ISSUE THREAD

**Issue by blimpich** [Expensify/App#46995]
# Problem

`AddBillingCardAndRequestPolicyOwnerChange` isn't 1:1:1.

# Solution

Make it 1:1:1.

<details><summary>Issue Owner</summary>Current Issue Owner: @blimpich</details>

**Comment by garrettmknight** [comment]
Adding BZ to monitor and take this one off hold when it's ready. Held on https://github.com/Expensify/App/issues/46994

**Comment by strepanier03** [comment]
Thanks Garret, I'll keep an eye on it.

**Comment by blimpich** [comment]
Making this weekly since its waiting on other issues to be done first.

**Comment by strepanier03** [comment]
#46994 is closed now. 

There is one remaining [PR](https://github.com/Expensify/App/issues/46994#issuecomment-2292444538) that we should wait for I think, then this can come off hold.

**Comment by blimpich** [comment]
Taking it off hold, this should be good to work on now. There's two parts to it:
1. need to move all [this logic](https://github.com/Expensify/Web-Expensify/blob/90aa78985c7b5136c8fd3e188137dad9458c4b58/lib/PolicyAPI.php#L4242-L4290) down into the Auth command `TakePolicyOwnership`
2. Make a new command in Auth called `AddBillingCardAndRequestPolicyOwnerChange` that calls the `CreateFund` command and then the `TakePolicyOwnership` command.

Could be split up into two different issues if the person who picks this up wants to. I currently don't have the bandwidth to pick this up though.

**Comment by blimpich** [comment]
Picking up, have bandwidth now

**Comment by blimpich** [comment]
Created new branches for working on this, wasn't able to spend much time on it today due to chores, but intend to get to this Tuesday/Wednesday this week.

I'll be OOO for most and potentially all of Thursday/Friday, so I estimate that I won't get a full PR out for this until sometime next week.

**Comment by blimpich** [comment]
Wasn't able to work on this today, got assigned a big chore to monitor API performance for the week. That took up most of my day. Hoping I'll get to this either tomorrow or Monday.

**Comment by blimpich** [comment]
There's a lot of different parts to this so I'm going to try and make many small PRs to complete this issue rather than one larger one. That way we also hopefully make it easier to review, and more likely that we'll find bugs before releasing to users.

**Comment by blimpich** [comment]
Lets make a checklist to break up what to move:
- [ ] [changeLog creation](https://github.com/Expensify/Web-Expensify/blob/77769295b97f7c06bfaf01042f1e74b4c7a7e4be/lib/PolicyAPI.php#L4235-L4250)
- [ ] [queue recommended account review job](https://github.com/Expensify/Web-Expensify/blob/77769295b97f7c06bfaf01042f1e74b4c7a7e4be/lib/PolicyAPI.php#L4269), not sure if this is actually even possible right now
- [ ] [transferAccountManager call](https://github.com/Expensify/Web-Expensify/blob/77769295b97f7c06bfaf01042f1e74b4c7a7e4be/lib/PolicyAPI.php#L4271)
- [ ] [notification to old owner](https://github.com/Expensify/Web-Expensify/blob/77769295b97f7c06bfaf01042f1e74b4c7a7e4be/lib/PolicyAPI.php#L4273-L4281)
- [ ] [retention whisper](https://github.com/Expensify/Web-Expensify/blob/77769295b97f7c06bfaf01042f1e74b4c7a7e4be/lib/PolicyAPI.php#L4283-L4302), also unsure if this one is possible but I will look into it

Today I worked on the first checkbox, made good progress.

**Comment by blimpich** [comment]
Spent time on this, got [the Auth PR](https://github.com/Expensify/Auth/pull/12512) out for review for the [first checkbox](https://github.com/Expensify/App/issues/46995#issuecomment-2344986615), waiting on that to merge before putting [web pr](https://github.com/Expensify/Web-Expensify/pull/43487) up for review

**Comment by blimpich** [comment]
Confirmed that we cannot in fact do checkbox number 2 and 5 because they are bedrock jobs and have no way of being queued from Auth. Crossed them off and created this new issue to add the job queues to the `verifySetupIntentAndRequestPolicyOwnerChange` web call.

**Comment by blimpich** [comment]
Calling it a day. PRs for the first checkbox are in review or are held waiting for dependent Auth PR. Will work on the remaining  checkboxes tomorrow.

**Comment by blimpich** [comment]
Couldn't make much progress on this, caught up with other chores/bugs.

**Comment by blimpich** [comment]
Looked into the 3rd checkbox, gets really complicated really quick but basically its also impossible to fully move over into Auth due to the fact that it calls `ChatBotAPI::updateDomainAccountManager` which queues jobs. Frustrating but so much of this command is just not possible to move into Auth right now. Checking off the third checkbox.

The 4th checkbox I know for a fact can be ported to Auth since we can queue notifications. So that will be the last thing I do for this issue.

**Comment by blimpich** [comment]
Waiting to get last bit of feedback on PR before merging, then this issue will be done: https://github.com/Expensify/Web-Expensify/pull/43595#discussion_r1771690613

**Comment by blimpich** [comment]
PR merged, waiting deployment / QA, but going to check off [last checkbox](https://github.com/Expensify/App/issues/46995#issuecomment-2344986615) right now
